architecture-and-base-libs.md

```diff
@@ -0,0 +1,117 @@
+
+Unless necessary, this chapter will not go into detail on the preferences that make up 
+the architecture. There are lots of amazing people who can explain these preferences 
+much better, repeating or copy pasting their words would be both a waste of time and 
+disservice to their work. So please check all the shared resources thoroughly, and 
+research more on your own if the resources provided are not enough, or you feel they 
+are not compatible with your current level of understanding.
+
+## Language Preferences
+
+We use TypeScript, because we want to suffer less by making the complier help us.
+
+Resources:
+
+[Typescript vs Javascript by Ben Award](https://www.youtube.com/watch?v=D6or2gdrHRE)
+
+Q:
+Wait! I don't know TypeScript and never used it before, what am I gonna do?
+
+A:
+Don't worry, just check some tutorials, start getting your hands dirty, and know 
+that your RN department teammates are always ready to help.
+
+## Components and Local State Management Preferences
+
+We use Functional Components and hooks.
+
+What is `Local State`?
+
+Local state can be considered as any state that is specific to a single component 
+and is not required to be shared across the application. The state of a smaller component
+in a bigger component (like a screen component (i.e. Settings), which has a button component 
+(i.e. Logout) that has isDisabled prop that is stored in Settings component) 
+can be also considered as local state, as long as we don't have to pass
+the props multiple levels down.
+
+Basic examples could be (let's imagine these are in useState hooks in a functional component):
+
+`[email, setEmail], [password, setPassword], [isLoginButtonActive, setIsLoginButtonActive]` in a login component.
+
+`[isProfileDetailHidden, setIsProfileDetailHidden]` in a Profile component.
+
+Resources:
+
+[Using React Hooks vs. Class Components by Ben Award](https://www.youtube.com/watch?v=vbaIZ3xMj9U)
+
+[React Function Components by Robin Wieruch](https://www.robinwieruch.de/react-function-component)
+
+[When to break up a component into multiple components](https://kentcdodds.com/blog/when-to-break-up-a-component-into-multiple-components)
+
+## Navigation
+
+We use [React Native Navigation](https://github.com/wix/react-native-navigation). 
+Our main reasoning is its performance compared to [React Navigation](https://reactnavigation.org/).
+
+## Global State Management Preferences
+
+We use [Redux](https://redux.js.org/introduction/getting-started) with 
+[hooks](https://react-redux.js.org/api/hooks), and [reselect](https://github.com/reduxjs/reselect) 
+library for complex selections.
+
+We use [Redux Thunk](https://github.com/reduxjs/redux-thunk) to write async 
+logic that interacts with the store.
+
+What is global state and why do I need it?
+
+Let's think our app as a top down tree, and each component user can navigate 
+to is a node on the branches of this tree. If we only had our state on top of 
```

Comment:
"to is" - a little bit confusing. In Ukrainian I can place a comma separator. Can we also separate this with "," or "-" ? Or rewrite the sentence somehow.